debezium-server/debezium-server-redis/src/test/java/io/debezium/server/redis/RedisStreamIT.java

```diff
@@ -61,15 +74,111 @@ void connectorCompleted(@Observes ConnectorCompletedEvent event) throws Exceptio
-        Testing.Print.enable();
+        // Verifies that all the records of a PostgreSQL table are streamed to Redi
+        final int MESSAGE_COUNT = 4;
+        final String STREAM_NAME = "testc.inventory.customers";
+
         final List<StreamEntry> entries = new ArrayList<>();
         Awaitility.await().atMost(Duration.ofSeconds(RedisTestConfigSource.waitForSeconds())).until(() -> {
             final List<StreamEntry> response = jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT);
+            entries.clear();
             entries.addAll(response);
             return entries.size() >= MESSAGE_COUNT;
         });
-        Assertions.assertThat(entries.size() >= MESSAGE_COUNT);
+        assertTrue("Redis Basic Stream Test Failed", entries.size() == MESSAGE_COUNT);
+    }
+
+    @Test
+    @FixFor("DBZ-4510")
+    public void testRedisConnectionRetry() throws Exception {
+        /*
+         * Test retry mechanism when encountering Redis connectivity issues:
+         * 1. Make Redis to be unavailable while the server is up
+         * 2. Create a new table named redis_test in PostgreSQL and insert 5 records to it
+         * 3. Bring Redis up again and make sure these records have been streamed successfully
+         */
+
+        final int MESSAGE_COUNT = 5;
+        final String STREAM_NAME = "testc.inventory.redis_test";
+        Testing.print("Pausing container");
+        RedisTestResourceLifecycleManager.pause();
+        final PostgresConnection connection = getPostgresConnection();
+        Testing.print("Creating new redis_test table and inserting 5 records to it");
+        connection.execute(
+                "CREATE TABLE inventory.redis_test (id INT PRIMARY KEY)",
+                "INSERT INTO inventory.redis_test VALUES (1)",
+                "INSERT INTO inventory.redis_test VALUES (2)",
+                "INSERT INTO inventory.redis_test VALUES (3)",
+                "INSERT INTO inventory.redis_test VALUES (4)",
+                "INSERT INTO inventory.redis_test VALUES (5)");
+        connection.close();
+
+        Testing.print("Sleeping for 5 seconds to simulate no connection errors");
+        Thread.sleep(5000);
+        Testing.print("Unpausing container");
+        RedisTestResourceLifecycleManager.unpause();
+
+        final List<StreamEntry> entries = new ArrayList<>();
+        Awaitility.await().atMost(Duration.ofSeconds(RedisTestConfigSource.waitForSeconds())).until(() -> {
+            final List<StreamEntry> response = jedis.xrange(STREAM_NAME, null, null, MESSAGE_COUNT);
+            entries.clear();
+            entries.addAll(response);
+            return entries.size() >= MESSAGE_COUNT;
+        });
+        Testing.print("Entries in " + STREAM_NAME + ":" + entries.size());
+        assertTrue("Redis Connection Test Failed", entries.size() == MESSAGE_COUNT);
+    }
+
+    @Test
+    @FixFor("DBZ-4510")
+    public void testRedisOOMRetry() throws Exception {
+        /*
```

Comment:
Same as above.

Reply:
Changed this based on your other comment.